lib/api/src/modules/stories.ts

```diff
@@ -239,14 +239,13 @@ export const init: ModuleFn = ({
         refs,
       } = store.getState();
 
-      const viewMode = viewModeFromArgs || viewModeFromState;
-
       const hash = ref ? refs[ref].stories : storiesHash;
 
       if (!story) {
         const s = hash[kindOrId] || hash[sanitize(kindOrId)];
         // eslint-disable-next-line no-nested-ternary
         const id = s ? (s.children ? s.children[0] : s.id) : kindOrId;
+        const viewMode = viewModeFromArgs || s ? s.parameters.viewMode : viewModeFromState;
```

Comment:
what if viewMode is not set in `s.parameters`?

Reply:
good call, will adjust